File: Cpp/1291_Sequential_Digits.cpp
class Solution {
public:
    vector<int> sequentialDigits(int low, int high) {
        vector<int> result;
        
        for(int i=1; i<10; ++i){
            int num = i;
            int next = i;
            while(num <= high && next < 10){
                if(num >= low){
                    result.push_back(num);
                }
                
                ++next;
                num = 10 * num + next;
            }
        }
        sort(result.begin(), result.end());
        
        return result;
    }
};

File: Kotlin/1291_Sequential_Digits.kt
class Solution {
    fun sequentialDigits(low: Int, high: Int): List<Int> {
        val result = mutableListOf<Int>()
        
        for (i in 1 until 10) {
            var num = i
            var nextNum = num
            while(num <= high && nextNum < 10){
                if(num >= low) result.add(num)
                ++nextNum
                num = 10 * num + nextNum
            }
        }
        
        return result.sorted()
    }
}